### PR TITLE
Send rate limited requests to BigQuery

### PR DESCRIPTION
**Pull request by ethax-ross**
[Jira-2596](https://dfedigital.atlassian.net/browse/CPDLP-2596)

### Context

We want to ensure web requests that are rate limited by Rack::Attack still get sent to BigQuery as a web request.

### Changes proposed in this pull request

- Send rate limited requests to BigQuery

Web requests are sent to BigQuery via a concern in the ApplicationController. If Rack intercepts the request it won't reach the controller, so we need to manually send web requests that are rate limited to BigQuery.

### Guidance to review

You can check 429s come through correctly in BigQuery:

```
SELECT * FROM `ecf-bq.ecf_events_review.events` WHERE response_status = "429" AND TIMESTAMP_TRUNC(occurred_at, DAY) = TIMESTAMP("2023-11-22") LIMIT 1000
```

You can generate more using something like `wrk`:

```
wrk -t5 -c5 -d5s https://cpd-ecf-review-4272-web.test.teacherservices.cloud
```

I tested by generating a bunch of fake traffic and Sidekiq had no problem staying on top of the jobs, I don't think there would be any issues shipping this to prod.

### CODE REVIEW

File: config/initializers/rack_attack.rb [hunk 1]
```diff
@@ -52,8 +52,25 @@ def remote_ip
...
+  # to manually send web requests that are rate limited to BigQuery.
+  rate_limit_event = DfE::Analytics::Event.new
+    .with_type(:web_request)
+    .with_request_details(request)
+    .with_response_details(response)
+    .with_request_uuid(request.uuid)
+
+  rate_limit_event.with_user(user) if user.respond_to?(:id)
+
+  DfE::Analytics::SendEvents.do([rate_limit_event])
```
Comment (cwrw) [inline config/initializers/rack_attack.rb:75]: could we only send this if `DfE::Analytics.enabled?`

**Review by cwrw — APPROVED** [review]
looks good, past the minor comment, can we test this with API requests from both lead provider and NPQ to make sure all is well just incase?

**Comment by ethax-ross** [comment]
Screenshot 2023-11-23 at 10 30 38

@cwrw I've beefed up the testing and manually checked it rate limits NPQ reg tokens

**Review by cwrw — APPROVED** [review]
Thanks @ethax-ross 🙌

**Pull request merged**